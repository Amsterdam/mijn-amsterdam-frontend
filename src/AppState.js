import React, { createContext } from 'react';
import { useBrpApi } from 'hooks/brp-api.hook.js';
import useSessionApi from 'hooks/session.api.hook.js';
import useMijnUpdatesApi from 'hooks/mijn-updates-api.hook';
import useMyCasesApi from 'hooks/my-cases-api.hook';
import useMyTipsApi from 'hooks/my-tips-api.hook';

export const AppContext = createContext();

export default function AppState({ children, value }) {
  // State that needs to be shared across multiple components.
  const appState = value || {
    BRP: useBrpApi(),
    SESSION: useSessionApi(),

    // NOTE: If needed we can postpone immediate fetching of below data and start fetching in the component.
    MY_UPDATES: useMijnUpdatesApi(),
    MY_CASES: useMyCasesApi(),
    MY_TIPS: useMyTipsApi(),
  };

  return <AppContext.Provider value={appState}>{children}</AppContext.Provider>;
}
