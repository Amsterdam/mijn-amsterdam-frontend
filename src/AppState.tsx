import React, { createContext } from 'react';
import { useBrpApi, BrpApiState } from 'hooks/api/brp-api.hook';
import useSessionApi, { SessionApiState } from 'hooks/api/session.api.hook';
import useMyUpdatesApi from 'hooks/api/my-updates-api.hook';
import useMyCasesApi from 'hooks/api/my-cases-api.hook';
import useMyTipsApi from 'hooks/api/my-tips-api.hook';
import { ComponentChildren } from './App.types';
import { MyUpdatesApiState } from './hooks/api/my-updates-api.hook';
import { MyCasesApiState } from './hooks/api/my-cases-api.hook';
import { MyTipsApiState } from './hooks/api/my-tips-api.hook';
import useWmoApi, { WmoApiState } from './hooks/api/api.wmo';
import useFocusApi, { FocusApiState } from './hooks/api/api.focus';
import useMyChapters from 'hooks/myChapters.hook';
import { MenuItem } from './components/MainNavBar/MainNavBar.constants';
import useErfpachtApi, { ErfpachtApiState } from './hooks/api/api.erfpacht';

export interface AppState {
  BRP: BrpApiState;
  SESSION: SessionApiState;
  MY_UPDATES: MyUpdatesApiState;
  MY_CASES: MyCasesApiState;
  MY_TIPS: MyTipsApiState;
  WMO: WmoApiState;
  FOCUS: FocusApiState;
  MY_CHAPTERS: MenuItem[];
  ERFPACHT: ErfpachtApiState;
}

// Use typecasting here to allow for proper state completion and use in deconstruction assignments.
export const AppContext = createContext<AppState>({} as AppState);
export const SessionContext = createContext<SessionApiState>(
  {} as SessionApiState
);

interface SessionStateProps {
  render: (session: SessionApiState) => ComponentChildren;
}

export function SessionState({ render }: SessionStateProps) {
  const session = useSessionApi();
  return (
    <SessionContext.Provider value={session}>
      {render(session)}
    </SessionContext.Provider>
  );
}

interface AppStateProps {
  children?: ComponentChildren;
  value?: Partial<AppState>;
  session?: SessionApiState;
  render?: (state: AppState) => ComponentChildren;
}

export default ({ render, children, value, session }: AppStateProps) => {
  let appState;

  if (typeof value !== 'undefined') {
    appState = value;
  } else {
    const WMO = useWmoApi();
    const FOCUS = useFocusApi();
    const BRP = useBrpApi();
    const MY_UPDATES = useMyUpdatesApi();
    const MY_CASES = useMyCasesApi();
    const MY_TIPS = useMyTipsApi();
    const ERFPACHT = useErfpachtApi();
    const MY_CHAPTERS = useMyChapters({ WMO, FOCUS, ERFPACHT });

    appState = {
      BRP,
      SESSION: session,

      // NOTE: If needed we can postpone immediate fetching of below data and start fetching in the component
      // by calling the refetch method implemented in the api hooks.
      MY_UPDATES,
      MY_CASES,
      MY_TIPS,
      WMO,
      FOCUS,
      MY_CHAPTERS,
      ERFPACHT,
    };
  }

  return (
    // TODO: Straight out partial appState assignments. Forcing type assignment here for !!!111!!1!!
    <AppContext.Provider value={appState as AppState}>
      {render ? render(appState as AppState) : children}
    </AppContext.Provider>
  );
};
