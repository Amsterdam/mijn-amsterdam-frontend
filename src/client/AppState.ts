import { createContext } from 'react';

import type {
  ServiceID,
  ServicesTips,
  ServicesType,
} from '../server/services/controller';
import { FeatureToggle } from '../universal/config';
import { apiPristineResult, ApiResponse } from '../universal/helpers/api';

export type AppState = {
  [key in ServiceID]: ApiResponse<
    ReturnTypeAsync<ServicesType[key]>['content']
  >;
} & {
  TIPS: ServicesTips | ApiResponse<any>;
};

export const PRISTINE_APPSTATE: AppState = {
  // Generated
  TIPS: apiPristineResult([]),
  NOTIFICATIONS: apiPristineResult([]),
  CASES: apiPristineResult([]),

  // Direct
  FOCUS_SPECIFICATIES: apiPristineResult({
    jaaropgaven: [],
    uitkeringsspecificaties: [],
  }),
  KREFIA: apiPristineResult(null, FeatureToggle.krefiaActive),
  FOCUS_AANVRAGEN: apiPristineResult([]),
  FOCUS_TOZO: apiPristineResult([]),
  FOCUS_TONK: apiPristineResult([]),
  FOCUS_BBZ: apiPristineResult([]),
  FOCUS_STADSPAS: apiPristineResult(null),
  WMO: apiPristineResult([]),
  ERFPACHT: apiPristineResult({ isKnown: false }),
  BELASTINGEN: apiPristineResult({ isKnown: true }),
  MILIEUZONE: apiPristineResult({ isKnown: false }),
  AKTES: apiPristineResult([], FeatureToggle.aktesActive),
  TOERISTISCHE_VERHUUR: apiPristineResult({
    vergunningen: [],
    registraties: [],
    daysLeft: 0,
  }),
  VERGUNNINGEN: apiPristineResult([]),

  // KVK / Handelsregister
  KVK: apiPristineResult(null),

  // Related
  BRP: apiPristineResult(null),
  AFVAL: apiPristineResult([]),
  AFVALPUNTEN: apiPristineResult({
    centers: [],
    datePublished: '',
  }),
  MY_LOCATION: apiPristineResult(null),

  // CMS content
  CMS_CONTENT: apiPristineResult({
    generalInfo: null,
    footer: null,
  }),
  CMS_MAINTENANCE_NOTIFICATIONS: apiPristineResult([]),
};

export const ALL_ERROR_STATE_KEY = 'ALL';

export function createAllErrorState(state: AppState, message: string) {
  return Object.assign({}, state, {
    [ALL_ERROR_STATE_KEY]: {
      status: 'ERROR',
      message,
    },
  });
}

export const AppContext = createContext<AppState>(PRISTINE_APPSTATE);
