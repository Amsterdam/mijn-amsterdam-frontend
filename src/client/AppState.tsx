import React, { createContext } from 'react';
import {
  ServicesDirectData,
  useServicesDirect,
} from './hooks/api/api.services-direct';
import { ServicesMapData, useServicesMap } from './hooks/api/api.services-map';
import {
  ServicesRelatedData,
  useServicesRelated,
} from './hooks/api/api.services-related';
import {
  ServicesTipsData,
  useServicesTips,
} from './hooks/api/api.services-tips';
import {
  ServicesUpdatesData,
  useServicesUpdates,
} from './hooks/api/api.services-updates';

import { ChaptersState } from './config/myChapters';
import { ComponentChildren } from '../universal/types/App.types';

export interface AppState extends ServicesRelatedData, ServicesDirectData {
  TIPS: ServicesTipsData;
  CHAPTERS: ChaptersState;
  BUURT: ServicesMapData;
  UPDATES: ServicesUpdatesData;
}
export type StateKey = keyof AppState;

export const AppContext = createContext<AppState>({} as AppState);

interface AppStateProps {
  children: ComponentChildren;
}

interface MockAppStateProps {
  children: ComponentChildren;
  value: Partial<AppState>;
}

export function useAppState(): AppState {
  const servicesRelatedApi = useServicesRelated();
  const servicesDirectApi = useServicesDirect();
  const tipsApi = useServicesTips();
  const updatesApi = useServicesUpdates();
  const mapData = useServicesMap();

  return {
    ...servicesRelatedApi.data,
    ...servicesDirectApi.data,
    TIPS: tipsApi.data,
    UPDATES: updatesApi.data,
    CHAPTERS: {
      isLoading: true,
      items: [],
    },
    BUURT: mapData,
  };
}

export function MockAppStateProvider({ children, value }: MockAppStateProps) {
  return (
    <AppContext.Provider value={value as AppState}>
      {children}
    </AppContext.Provider>
  );
}

export default function AppStateProvider({ children }: AppStateProps) {
  const appState = useAppState();
  return <AppContext.Provider value={appState}>{children}</AppContext.Provider>;
}
