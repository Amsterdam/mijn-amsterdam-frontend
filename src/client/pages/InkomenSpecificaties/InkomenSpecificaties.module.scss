.SpecificationsTable {
  tr td:first-child {
    font-weight: 500;
    color: $color-black;
  }
  tr td:nth-child(2) {
    width: 30%;
  }
  @include media-q('>=phone') {
    tr td:last-of-type {
      width: 70px;
      text-align: right;
    }
  }
  @include select(Table_DisplayPropTitle) {
    width: 30%;
  }
}

.SpecificationsTable--annualStatements {
  @include select(Table_DisplayPropTitle) {
    width: 60%;
  }
  tr td:nth-child(2) {
    width: calc(30% - 1em);
  }
}

.SearchButton {
  margin-bottom: 2rem;

  @include media-q('>=phone') {
    float: right;
  }

  i {
    height: 1.3rem;
    font-style: normal;
    position: relative;
    transform: rotate(45deg) translateY(-0.6rem) scale(0.6);
    transform-origin: center;
  }

  &Active {
    i {
      transform: rotate(-135deg) translateY(-0.2rem) scale(0.6)
        translateX(-0.8rem);
    }
  }
}

.SearchResetButton {
}

.SearchPanel {
  background-color: $color-neutral-grey1;
  width: 100%;
  display: flex;
  padding: 2rem;
  margin-bottom: 2rem;
  flex-wrap: wrap;

  @include media-q('<=phone') {
    flex-direction: column;
  }
}

.Pagination {
  margin: 0 auto;
  margin-top: 4rem;
  width: 50%;
  display: flex;
  justify-content: center;
}

.FilterInput {
  display: flex;
  flex-direction: column;
  flex: 2;

  > span {
    display: inline-block;
    margin-bottom: 3px;
  }

  @include media-q('>phone') {
    &:first-child {
      margin-right: 2rem;
    }

    &:last-child {
      margin-left: 2rem;
    }
  }
}

.Select {
  @include inputSelect;
}

.FilterActive {
  border-color: $color-primary-darkblue;
  outline: 1px solid $color-primary-darkblue;
  color: $color-primary-darkblue;
}

.ResetFilterButton {
  border: 0;
  background: transparent;
  padding: 0;
  width: auto;
  text-decoration: underline;
}
