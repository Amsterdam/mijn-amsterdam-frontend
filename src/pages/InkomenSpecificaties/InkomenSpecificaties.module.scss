@import 'styles/_global.scss';

.SpecificationsTable {
  tr td:first-child {
    font-weight: 500;
    color: $color-black;
  }
  tr td:nth-child(2) {
    width: 30%;
  }
  @include media-q('>=phone') {
    tr td:last-of-type {
      width: 70px;
      text-align: right;
    }
  }
  @include select(Table_DisplayPropTitle) {
    width: 30%;
  }
}

.SpecificationsTable--annualStatements {
  @include select(Table_DisplayPropTitle) {
    width: calc(60% + 30px);
    padding-right: 2rem;
  }
}

.TableSection {
  @include centered-block;
  position: relative;

  @include select(Heading_Heading) {
    margin: 5rem auto;
    max-width: 62rem;

    @include mq-tablet {
      margin-left: 9.5rem;
    }

    @include mq-phone {
      margin-left: 0;
    }
  }
}

.SearchButton {
  margin-bottom: 2rem;

  @include media-q('>=phone') {
    float: right;
  }

  svg {
    height: 1.3rem;
    fill: $color-white;
  }


}

.SearchResetButton {

}

.SearchPanel {
  background-color: $color-neutral-grey1;
  width: 100%;
  display: flex;
  padding: 2rem;
  margin-bottom: 2rem;
  flex-wrap: wrap;

  @include media-q('<=phone') {
    flex-direction: column;
  }

  label {
    display: flex;
    flex-direction: column;
    flex: 2;

    @include media-q('>phone') {
      &:first-child {
        margin-right: 2rem;
      }

      &:last-child {
        margin-left: 2rem;
      }
    }
  }
}

.Pagination {
  margin: 0 auto;
  margin-top: 4rem;
  width: 50%;
  display: flex;
  justify-content: center;
}

.Select {
  @include inputSelect;
}
