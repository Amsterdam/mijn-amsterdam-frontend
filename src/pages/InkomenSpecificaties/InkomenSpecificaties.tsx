import DateInput, {
  isNativeDatePickerInputSupported,
} from 'components/DateInput/DateInput';
import { OverviewPage, PageContent } from 'components/Page/Page';
import React, {
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';

import Alert from 'components/Alert/Alert';
import { AppContext } from 'AppState';
import { AppRoutes } from 'config/Routing.constants';
import { Button } from 'components/Button/Button';
import ChapterIcon from 'components/ChapterIcon/ChapterIcon';
import { ChapterTitles } from 'config/Chapter.constants';
import PageHeading from 'components/PageHeading/PageHeading';
import Pagination from 'components/Pagination/Pagination';
import Section from 'components/Section/Section';
import Table from 'components/Table/Table';
import classnames from 'classnames';
import styles from './InkomenSpecificaties.module.scss';
import useRouter from 'use-react-router';
import { parseISO } from 'date-fns';

export const specificationsTableDisplayProps = {
  title: 'Omschrijving',
  type: 'Regeling',
  displayDate: 'Datum',
  documentUrl: 'Documenten',
};

export const annualStatementsTableDisplayProps = {
  title: 'Omschrijving',
  displayDate: 'Datum',
  documentUrl: 'Documenten',
};

const PAGE_SIZE = 10;
const INITIAL_INDEX = [0, PAGE_SIZE - 1];

function Caret() {
  return <i className={styles.SearchButtonIcon}>&#9698;</i>;
}

export default () => {
  const {
    FOCUS_INKOMEN_SPECIFICATIES: {
      data: { jaaropgaven, uitkeringsspecificaties },
      isError,
      isLoading,
    },
  } = useContext(AppContext);

  const {
    match: {
      params: { type },
    },
  } = useRouter();

  const isAnnualStatementOverviewPage = type === 'jaaropgaven';

  const items = isAnnualStatementOverviewPage
    ? jaaropgaven
    : uitkeringsspecificaties;

  const maxDate = useMemo(() => {
    if (items.length) {
      return parseISO(items[0].datePublished);
    }
    return new Date();
  }, [items]);

  const minDate = useMemo(() => {
    if (items.length) {
      return parseISO(items[items.length - 1].datePublished);
    }
    return new Date();
  }, [items]);

  const [isSearchPanelActive, setSearchPanelActive] = useState(false);
  const [selectedType, setSelectedType] = useState('');
  const [selectedDates, setSelectedDates] = useState<[Date, Date]>([
    minDate,
    maxDate,
  ]);

  useEffect(() => {
    // Set initial date selection
    if (minDate && maxDate) {
      setSelectedDates([minDate, maxDate]);
    }
  }, [minDate, maxDate]);

  const options: Array<[string, number]> = useMemo(() => {
    return Array.from(
      items.reduce((acc: Map<string, number>, item) => {
        acc.set(item.type, (acc.get(item.type) || 0) + 1);
        return acc;
      }, new Map<string, number>())
    );
  }, [items]);

  const [[startIndex, endIndex], setPageIndex] = useState(INITIAL_INDEX);

  const itemsFiltered = items
    .filter(item => (selectedType ? item.type === selectedType : true))
    .filter(item => {
      const datePublished = parseISO(item.datePublished);
      return (
        datePublished >= selectedDates[0] && datePublished <= selectedDates[1]
      );
    });

  const itemsFilteredPaginated = itemsFiltered.slice(startIndex, endIndex + 1);
  const typeFilterActive = !!selectedType;
  const minDateFilterActive =
    selectedDates[0].toString() !== minDate.toString();
  const maxDateFilterActive =
    selectedDates[1].toString() !== maxDate.toString();

  const selectTypeFilter = useCallback(type => {
    setSelectedType(type);
    setPageIndex(INITIAL_INDEX);
  }, []);

  function resetSearch() {
    setSelectedType('');
    setSelectedDates([minDate, maxDate]);
  }

  function toggleSearchPanel() {
    const isActive = !isSearchPanelActive;
    setSearchPanelActive(isActive);
    if (!isActive) {
      resetSearch();
    }
  }

  return (
    <OverviewPage
      className={classnames(
        styles.InkomenSpecificaties,
        isAnnualStatementOverviewPage &&
          styles['SpecificationsTable--annualStatements']
      )}
    >
      <PageHeading
        icon={<ChapterIcon />}
        backLink={{ to: AppRoutes.INKOMEN, title: ChapterTitles.INKOMEN }}
      >
        Bijstandsuitkering {isAnnualStatementOverviewPage && 'Jaaropgaven'}
      </PageHeading>
      <PageContent>
        {isError && (
          <Alert type="warning">
            <p>We kunnen op dit moment niet alle gegevens tonen.</p>
          </Alert>
        )}
      </PageContent>
      <Section
        className={styles.TableSection}
        title={
          isAnnualStatementOverviewPage
            ? 'Jaaropgaven'
            : 'Uitkeringsspecificaties'
        }
        isLoading={isLoading}
        hasItems={!!items.length}
        noItemsMessage={
          'Er zijn op dit moment nog geen documenten beschikbaar.'
        }
      >
        <Button
          variant="secondary-inverted"
          className={classnames(
            styles.SearchButton,
            isSearchPanelActive && styles.SearchButtonActive
          )}
          onClick={toggleSearchPanel}
          disabled={isSearchPanelActive}
          icon={Caret}
          iconPosition="right"
        >
          Zoeken
        </Button>

        {isSearchPanelActive && (
          <div className={styles.SearchPanel}>
            {items.some(item => !!item.type) && (
              <div className={styles.FilterInput}>
                <span>
                  Regeling{' '}
                  {typeFilterActive && (
                    <button
                      className={styles.ResetFilterButton}
                      onClick={() => setSelectedType('')}
                    >
                      resetten
                    </button>
                  )}
                </span>
                <select
                  className={classnames(
                    styles.Select,
                    typeFilterActive && styles.FilterActive
                  )}
                  value={selectedType}
                  onChange={event => selectTypeFilter(event.target.value)}
                >
                  <option value="">Alle regelingen ({items.length})</option>
                  {options.map(([option, count]) => (
                    <option key={option} value={option}>
                      {option} ({count})
                    </option>
                  ))}
                </select>
              </div>
            )}
            <div className={styles.FilterInput}>
              <span>
                Datum van{' '}
                {minDateFilterActive && (
                  <button
                    className={styles.ResetFilterButton}
                    onClick={() =>
                      setSelectedDates([minDate, selectedDates[1]])
                    }
                  >
                    resetten
                  </button>
                )}
              </span>
              <DateInput
                className={classnames(
                  minDateFilterActive && styles.FilterActive
                )}
                value={selectedDates[0]}
                hasNativeSupport={isNativeDatePickerInputSupported()}
                onChange={dateStart => {
                  setSelectedDates(([, dateEnd]) => [
                    dateStart || minDate,
                    dateEnd || maxDate,
                  ]);
                }}
              />
            </div>
            <div className={styles.FilterInput}>
              <span>
                Datum t/m{' '}
                {maxDateFilterActive && (
                  <button
                    className={styles.ResetFilterButton}
                    onClick={() => {
                      setSelectedDates([selectedDates[0], maxDate]);
                    }}
                  >
                    resetten
                  </button>
                )}
              </span>
              <DateInput
                className={classnames(
                  maxDateFilterActive && styles.FilterActive
                )}
                value={selectedDates[1]}
                hasNativeSupport={isNativeDatePickerInputSupported()}
                onChange={dateEnd =>
                  setSelectedDates(([dateStart]) => [
                    dateStart || minDate,
                    dateEnd || maxDate,
                  ])
                }
              />
            </div>
          </div>
        )}
        {!itemsFiltered.length && (
          <p>
            Zoeken heeft geen resultaten opgeleverd.{' '}
            <Button
              onClick={resetSearch}
              variant="inline"
              className={styles.ResetButton}
            >
              Resetten
            </Button>
          </p>
        )}
        {!!itemsFiltered.length && (
          <Table
            className={styles.SpecificationsTable}
            items={itemsFilteredPaginated}
            displayProps={
              isAnnualStatementOverviewPage
                ? annualStatementsTableDisplayProps
                : specificationsTableDisplayProps
            }
          />
        )}

        {itemsFiltered.length > PAGE_SIZE && (
          <Pagination
            className={styles.Pagination}
            totalCount={itemsFiltered.length}
            pageSize={PAGE_SIZE}
            onPageClick={(page, ...index) => setPageIndex(index)}
          />
        )}
      </Section>
    </OverviewPage>
  );
};
