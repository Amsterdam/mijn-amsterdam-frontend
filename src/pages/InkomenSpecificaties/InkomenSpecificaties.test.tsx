import React from 'react';
import { shallow } from 'enzyme';
import InkomenSpecificaties from './InkomenSpecificaties';
import AppState from 'AppState';

const APP_STATE = {}; // Add slice of the AppState here

it('Renders without crashing', () => {
  shallow(
    <AppState value={APP_STATE}>
      <InkomenSpecificaties />
    </AppState>
  );
});
