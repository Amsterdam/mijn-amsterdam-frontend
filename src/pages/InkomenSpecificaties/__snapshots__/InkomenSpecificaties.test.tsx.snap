// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InkomenSpecificaties /> Renders a list with Jaaropgaven items 1`] = `"<main class=\\"Page OverviewPage InkomenSpecificaties\\"><header class=\\"PageHeading HasIcon\\"><span aria-hidden=\\"true\\" class=\\"Icon\\"><svg aria-label=\\"ROOT\\" fill=\\"black\\" class=\\"ChapterIcon\\">burgerzaken.svg</svg></span><div class=\\"HeadingInner\\"><a class=\\"Button Button__plain Button__lean Linkd BackLink\\" href=\\"/werk-en-inkomen\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" class=\\"Icon Icon__left\\">Chevron-Left.svg</svg>Inkomen en Stadspas</a><h2 class=\\"Heading Heading__large\\">Bijstandsuitkering Jaaropgaven</h2></div></header><div class=\\"PageContent\\"></div><section class=\\"Section TableSection\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\">Jaaropgaven</h3><button class=\\"Button Button__secondary SearchButton\\">Zoeken<svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" class=\\"Icon Icon__right\\">Search.svg</svg></button><table class=\\"Table SpecificationsTable\\"><thead><tr class=\\"TableRow\\"><th class=\\"DisplayProp\\">Omschrijving</th><th class=\\"DisplayProp\\">Type</th><th class=\\"DisplayProp\\">Datum</th><th class=\\"DisplayProp\\"></th></tr></thead><tbody><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Jaaropgave</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Type:</span>BIBI</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span>06 maart 2016</td><td class=\\"DisplayProp\\"><a href=\\"/api/focus/document?id=x\\" rel=\\"external noopener noreferrer\\" class=\\"DownloadLink\\" download=\\"2016-03-06-Jaaropgave\\"><svg width=\\"14\\" height=\\"14\\">Document.svg</svg> PDF</a></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Jaaropgave</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Type:</span>BIBI</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span>06 maart 2016</td><td class=\\"DisplayProp\\"><a href=\\"/api/focus/document?id=x\\" rel=\\"external noopener noreferrer\\" class=\\"DownloadLink\\" download=\\"2016-03-06-Jaaropgave\\"><svg width=\\"14\\" height=\\"14\\">Document.svg</svg> PDF</a></td></tr></tbody></table></section></main>"`;

exports[`<InkomenSpecificaties /> Renders a list with Uitkeringspecificaties items 1`] = `"<main class=\\"Page OverviewPage InkomenSpecificaties\\"><header class=\\"PageHeading HasIcon\\"><span aria-hidden=\\"true\\" class=\\"Icon\\"><svg aria-label=\\"ROOT\\" fill=\\"black\\" class=\\"ChapterIcon\\">burgerzaken.svg</svg></span><div class=\\"HeadingInner\\"><a class=\\"Button Button__plain Button__lean Linkd BackLink\\" href=\\"/werk-en-inkomen\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" class=\\"Icon Icon__left\\">Chevron-Left.svg</svg>Inkomen en Stadspas</a><h2 class=\\"Heading Heading__large\\">Bijstandsuitkering </h2></div></header><div class=\\"PageContent\\"></div><section class=\\"Section TableSection\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\">Uitkeringsspecificaties</h3><button class=\\"Button Button__secondary SearchButton\\">Zoeken<svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" class=\\"Icon Icon__right\\">Search.svg</svg></button><table class=\\"Table SpecificationsTable\\"><thead><tr class=\\"TableRow\\"><th class=\\"DisplayProp\\">Omschrijving</th><th class=\\"DisplayProp\\">Type</th><th class=\\"DisplayProp\\">Datum</th><th class=\\"DisplayProp\\"></th></tr></thead><tbody><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Type:</span>STIMREG</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span>06 juni 2016</td><td class=\\"DisplayProp\\"><a href=\\"/api/focus/document?id=x\\" rel=\\"external noopener noreferrer\\" class=\\"DownloadLink\\" download=\\"2016-06-06-Uitkeringspecificatie\\"><svg width=\\"14\\" height=\\"14\\">Document.svg</svg> PDF</a></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Type:</span>BBS</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span>06 april 2016</td><td class=\\"DisplayProp\\"><a href=\\"/api/focus/document?id=x\\" rel=\\"external noopener noreferrer\\" class=\\"DownloadLink\\" download=\\"2016-04-06-Uitkeringspecificatie\\"><svg width=\\"14\\" height=\\"14\\">Document.svg</svg> PDF</a></td></tr></tbody></table></section></main>"`;
